### PR TITLE
nRF54L20 board soc support

### PR DESCRIPTION
**Pull request by kl-cruz**
Added support for nRF54L20 device with configuration of initial PDK board.

### CODE REVIEW

File: boards/nordic/nrf54l20pdk/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
++-----------+------------+----------------------+
+| RRAM      | on-chip    | flash                |
++-----------+------------+----------------------+
+| GPIO      | on-chip    | gpio                 |
++-----------+------------+----------------------+
+| MPU       | on-chip    | arch/arm             |
++-----------+------------+----------------------+
+| NVIC      | on-chip    | arch/arm             |
++-----------+------------+----------------------+
+| GRTC      | on-chip    | counter              |
```
Comment (hubertmis) [inline boards/nordic/nrf54l20pdk/doc/index.rst:57]: Is GRTC used by a counter driver? I would expect a timer driver instead. I think TIMERs are used by the counter driver

File: boards/nordic/nrf54l20pdk/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
++-----------+------------+----------------------+
+| SPU       | on-chip    | system protection    |
```
Comment (hubertmis) [inline boards/nordic/nrf54l20pdk/doc/index.rst:59]: what is the key of including peripherals in this table? It seems strange to include `SPU` and exclude `MPC` - they have similar role in the system (configuration of memory bus access protection).

File: boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+&hfxo {
+	load-capacitors = "internal";
+	load-capacitance-femtofarad = <15000>;
+};
+
+&grtc {
+	owned-channels = <0 1 2 3 4 5 6 7 8 9 10 11>;
+	/* Channels 7-11 reserved for Zero Latency IRQs, 3-4 for FLPR */
+	child-owned-channels = <3 4 7 8 9 10 11>;
```
Comment (hubertmis) [inline boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi:42]: Side note: we need another way of expressing channel reservations for ZLI. This is confusing.

File: boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		};
+		slot1_partition: partition@b2000 {
+			label = "image-1";
+			reg = <0xb2000 DT_SIZE_K(324)>;
+		};
+		slot1_ns_partition: partition@103000 {
+			label = "image-1-nonsecure";
+			reg = <0x103000 DT_SIZE_K(324)>;
+		};
+		/* 32k from 0x154000 to 0x15bfff reserved for TF-M partitions */
```
Comment (hubertmis) [inline boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi:71]: It looks TF-M should use `slot0_partition` and `slot1_partition`. What is the purpose of reserving additional space?

File: boards/nordic/nrf54l20pdk/nrf54l20pdk_nrf54l20_cpuapp_defconfig [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+# MPU-based null-pointer dereferencing detection cannot
+# be applied as the (0x0 - 0x400) is unmapped for this target.
+CONFIG_NULL_POINTER_EXCEPTION_DETECTION_NONE=y
+
+# Enable Cache
+CONFIG_CACHE_MANAGEMENT=y
+CONFIG_EXTERNAL_CACHE=y
+
+CONFIG_SOC_NRF_FORCE_CONSTLAT=y
```
Comment (hubertmis) [inline boards/nordic/nrf54l20pdk/nrf54l20pdk_nrf54l20_cpuapp_defconfig:28]: Is this entry necessary? It's name suggests it disables power management in the device.

File: boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+/ {
+	chosen {
+		zephyr,console = &uart20;
+		zephyr,shell-uart = &uart20;
+		zephyr,uart-mcumgr = &uart20;
+		zephyr,bt-mon-uart = &uart20;
+		zephyr,bt-c2h-uart = &uart20;
+		zephyr,flash-controller = &rram_controller;
+		zephyr,flash = &cpuapp_rram;
+		zephyr,ieee802154 = &ieee802154;
```
Comment (hubertmis) [inline boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi:21]: I'm curious why `ieee802154` is selected at the board level while `bt-hci` is chosen at the soc level. Should we aim to unify it?

File: dts/common/nordic/nrf54l20.dtsi [hunk 7]
```diff
@@ -0,0 +1,614 @@
...
+			reg = <0x2002f000 DT_SIZE_K(68)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2002f000 0x11000>;
+		};
+
+#ifdef USE_NON_SECURE_ADDRESS_MAP
+		global_peripherals: peripheral@40000000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
```
Comment (hubertmis) [inline dts/common/nordic/nrf54l20.dtsi:99]: nit: these properties could be moved out of the `#ifdef` block. They are the same for all branches.

File: dts/common/nordic/nrf54l20.dtsi [hunk 8]
```diff
@@ -0,0 +1,614 @@
...
+				compatible = "nordic,nrf-vpr-coprocessor";
+				reg = <0x4c000 0x1000>;
+				ranges = <0x0 0x4c000 0x1000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				status = "disabled";
+
+				cpuflpr_clic: interrupt-controller@f0000000 {
+					compatible = "nordic,nrf-clic";
+					reg = <0xf0000000 0x1780>;
```
Comment (hubertmis) [inline dts/common/nordic/nrf54l20.dtsi:150]: Is this correct? If I'm not mistaken, this property is an offset from the `ranges` of its parent (0x4004c000) what gives 0x13004c000 - out of addressable memory.

I think clic should be defined outside of `global_peripherals` in a FLPR-specific file, as an IRQ controller at the `/soc/` level.

File: dts/common/nordic/nrf54l20.dtsi [hunk 9]
```diff
@@ -0,0 +1,614 @@
...
+
+			clock: clock@10e000 {
+				compatible = "nordic,nrf-clock";
+				reg = <0x10e000 0x1000>;
+				interrupts = <270 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+		};
+
+		rram_controller: rram-controller@5004e000 {
```
Comment (hubertmis) [inline dts/common/nordic/nrf54l20.dtsi:573]: It looks this node should be conditional with something like:

```
#ifdef USE_NON_SECURE_ADDRESS_MAP

rram_controller: rram-controller@4004e000 {

#else

rram_controller: rram-controller@5004e000 {
			compatible = "nordic,rram-controller";
			reg = <0x5004e000 0x1000>;
			interrupts = <75 NRF_DEFAULT_IRQ_PRIORITY>;

#endif

			#address-cells = <1>;
			#size-cells = <1>;
			cpuapp_rram: rram@0 {
				compatible = "soc-nv-flash";
...
```

The driver should be available only for secure firmware because this peripheral is accessible only from S mode (removed `compatible` for NS mode). Also, the address for S and NS modes should differ, though in NS mode it's probably not relevant as the driver is unused.

File: dts/common/nordic/nrf54l20.dtsi [hunk 10]
```diff
@@ -0,0 +1,614 @@
...
+				#interrupt-cells = <2>;
+			};
+
+			cpuapp_systick: timer@e000e010 {
+				compatible = "arm,armv8m-systick";
+				reg = <0xe000e010 0x10>;
+				status = "disabled";
+			};
+		};
+	};
```
Comment (hubertmis) [inline dts/common/nordic/nrf54l20.dtsi:613]: This section should be moved to a file like dts/arm/nordic/nrf54l20_cpuapp.dtsi. It should not be visible in FLPR builds.

File: dts/common/nordic/nrf54l20.dtsi (hunk 8 above)
Comment (kl-cruz) [reply dts/common/nordic/nrf54l20.dtsi:150]: As it is not supported for now, I will remove it

File: boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi (hunk 6 above)
Comment (kl-cruz) [reply boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi:21]: I based my code on nRF54L15. It can be question to someone else

File: dts/common/nordic/nrf54l20.dtsi (hunk 10 above)
Comment (kl-cruz) [reply dts/common/nordic/nrf54l20.dtsi:613]: I would prefer to keep it and unify description for nRF devices later

File: boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi (hunk 4 above)
Comment (kl-cruz) [reply boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi]: Removed

File: boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi (hunk 3 above)
Comment (kl-cruz) [reply boards/nordic/nrf54l20pdk/nrf54l20_cpuapp_common.dtsi:41]: Lets do it outside this integration

**Review by hubertmis — DISMISSED** [review]

File: boards/nordic/nrf54l20pdk/Kconfig [hunk 11]
```diff
@@ -0,0 +1,13 @@
+# nRF54L15 PDK board configuration
```
[the remark this answers is not in the document]
Comment (kl-cruz) [reply boards/nordic/nrf54l20pdk/Kconfig]: Done

File: boards/nordic/nrf54l20pdk/Kconfig [hunk 12]
```diff
@@ -0,0 +1,13 @@
+# nRF54L20 PDK board configuration
+
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_NRF54L20PDK_NRF54L20_CPUAPP
+
+config BOARD_ENABLE_DCDC
```
Comment (nordicjm) [inline boards/nordic/nrf54l20pdk/Kconfig:8]: @gmarull will your DCDC changes be ready soon?

File: boards/nordic/nrf54l20pdk/doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+* High-frequency 32 MHz crystal oscillator (HFXO)
+* Low-frequency 32.768 kHz crystal oscillator (LFXO)
+
+The crystal oscillators can be configured to use either
+internal or external capacitors.
+
+Supported Features
+==================
+
+The ``nrf54l20pdk/nrf54l20/cpuapp`` board configuration supports the following
```
Comment (nordicjm) [inline boards/nordic/nrf54l20pdk/doc/index.rst:38]: board target

File: boards/nordic/nrf54l20pdk/doc/index.rst [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
++-----------+------------+----------------------+
+| UARTE     | on-chip    | serial               |
++-----------+------------+----------------------+
+
+Other hardware features have not been enabled yet for this board.
+
+Programming and Debugging
+*************************
+
+Applications for the ``nrf54l20pdk/nrf54l20/cpuapp`` board can be
```
Comment (nordicjm) [inline boards/nordic/nrf54l20pdk/doc/index.rst:60]: board target

File: boards/nordic/nrf54l20pdk/nrf54l20pdk_nrf54l20-pinctrl.dtsi [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+		};
+	};
+
+	/omit-if-no-ref/ uart20_sleep: uart20_sleep {
+		group1 {
+			psels = <NRF_PSEL(UART_TX, 1, 4)>,
+				<NRF_PSEL(UART_RX, 1, 5)>;
+			low-power-enable;
+		};
+	};
```
Comment (nordicjm) [inline boards/nordic/nrf54l20pdk/nrf54l20pdk_nrf54l20-pinctrl.dtsi:23]: seems to conflict with the `hw-flow-control;` flag above

File: dts/arm/nordic/nrf54l20_cpuapp.dtsi [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+	status = "okay";
+};
+
+&cpuapp_ppb {
+	compatible = "simple-bus";
+	ranges;
+};
+
+&grtc {
+#ifdef USE_NON_SECURE_ADDRESS_MAP
```
Comment (nordicjm) [inline dts/arm/nordic/nrf54l20_cpuapp.dtsi:45]: Where is this symbol set, why is it not a Kconfig?

File: soc/nordic/nrf54l/Kconfig.defconfig.nrf54l20_enga_cpuapp [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+	default 271
+
+config IEEE802154_NRF5
+	default IEEE802154
+
+#config SOC_NRF54LX_SKIP_GLITCHDETECTOR_DISABLE
+#	default y
+#
+#config NFCT_PINS_AS_GPIOS
+#	default y
```
Comment (nordicjm) [inline soc/nordic/nrf54l/Kconfig.defconfig.nrf54l20_enga_cpuapp:18]: ?

File: dts/arm/nordic/nrf54l20_cpuapp.dtsi [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+	};
+};
+
+&bt_hci_controller {
+	status = "okay";
+};
+
+&ieee802154 {
+	status = "okay";
+};
```
Comment (nordicjm) [inline dts/arm/nordic/nrf54l20_cpuapp.dtsi:37]: why are both of these marked as okay in file in dts, should be board level enablement?

File: dts/arm/nordic/nrf54l20_cpuapp.dtsi (hunk 16 above)
Comment (kl-cruz) [reply dts/arm/nordic/nrf54l20_cpuapp.dtsi:45]: To be removed

File: dts/arm/nordic/nrf54l20_cpuapp.dtsi (hunk 18 above)
Comment (kl-cruz) [reply dts/arm/nordic/nrf54l20_cpuapp.dtsi:37]: To be moved

File: dts/arm/nordic/nrf54l20_cpuapp.dtsi [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+/ {
+	soc {
+		compatible = "simple-bus";
+		interrupt-parent = <&cpuapp_nvic>;
+		ranges;
+	};
+
+	psa_rng: psa-rng {
+		compatible = "zephyr,psa-crypto-rng";
+		status = "disabled";
```
Comment (e-rk) [inline dts/arm/nordic/nrf54l20_cpuapp.dtsi:22]: Is PSA RNG not functional at this time?
If yes, then would it make sense to temporarily add a PRNG node?
```
        prng: prng {
		compatible = "nordic,entropy-prng";
		status = "okay";
	};
```
Lack of the entropy node in the SoC DTSI makes it difficult to build many applications, because you basically need to make an application DTS overlays with the snippet I added everywhere.

If you go that route, you also need to add `SOC_SERIES_NRF54LX` to `depends` of  `FAKE_ENTROPY_NRF_PRNG` Kconfig and set the `zephyr,entropy`.

**Review by e-rk — APPROVED** [review]

File: boards/nordic/nrf54l20pdk/doc/index.rst [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+
+The nRF54L20 Preview Development Kit hardware provides
+support for the Nordic Semiconductor nRF54L20 Arm Cortex-M33 CPU and
+the following devices:
+
+* CLOCK
+* RRAM
+* :abbr:`GPIO (General Purpose Input Output)`
+* :abbr:`NVIC (Nested Vectored Interrupt Controller)`
+* :abbr:`UARTE (Universal asynchronous receiver-transmitter)`
```
Comment (anangl) [inline boards/nordic/nrf54l20pdk/doc/index.rst:22]: Shouldn't this list be a bit broader, more like the one for the nRF54L15 PDK? At least in `nrf54l20_cpuapp_common.dtsi` more stuff is enabled (`adc`, `grtc`).

**Review by anangl — CHANGES_REQUESTED** [review]
Please split this big commit into smaller ones related to particular areas: `soc:`, `dts:`, `modules:`, `boards:`, and a separate one that introduces `SOC_NRF54LX_SKIP_TAMPC_SETUP`.

File: boards/nordic/nrf54l20pdk/nrf54l20pdk_nrf54l20_cpuapp.yaml [hunk 21]
```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+identifier: nrf54l20pdk/nrf54l20/cpuapp
+name: nRF54l20-PDK-nRF54l20-Application
```
[the remark this answers is not in the document]
Comment (kl-cruz) [reply boards/nordic/nrf54l20pdk/nrf54l20pdk_nrf54l20_cpuapp.yaml:5]: we need to update L15 as well

File: modules/hal_nordic/nrfx/CMakeLists.txt [hunk 22]
```diff
@@ -47,6 +47,11 @@ zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54L15_ENGA_CPUAPP NRF_APPLICATION
 zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54L15_ENGA_CPUFLPR NRF_FLPR)
 zephyr_compile_definitions_ifdef(CONFIG_SOC_COMPATIBLE_NRF54L15 NRF54L15_XXAA)
 zephyr_compile_definitions_ifdef(CONFIG_SOC_COMPATIBLE_NRF54L15_CPUAPP NRF_APPLICATION)
+zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54L20_ENGA       NRF54L20_ENGA_XXAA)
+zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54L20_ENGA_CPUAPP NRF_APPLICATION)
+zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54L20_ENGA_CPUFLPR NRF_FLPR)
```
[the remark this answers is not in the document]
Comment (kl-cruz) [reply modules/hal_nordic/nrfx/CMakeLists.txt:52]: I forgot to remove it, thank you

**Comment by kl-cruz** [comment]
> Please split this big commit into smaller ones related to particular areas: `soc:`, `dts:`, `modules:`, `boards:`, and a separate one that introduces `SOC_NRF54LX_SKIP_TAMPC_SETUP`.

Done

**Review by anangl — DISMISSED** [review]

**Review by jaz1-nordic — DISMISSED** [review]

File: boards/nordic/nrf54l20pdk/Kconfig (hunk 12 above)
Comment (gmarull) [reply boards/nordic/nrf54l20pdk/Kconfig:8]: yes, please, wait until https://github.com/zephyrproject-rtos/zephyr/pull/74592 gets merged

**Review by gmarull — CHANGES_REQUESTED** [review]
see my comment

**Comment by kl-cruz** [comment]
> see my comment

I see it is merged now

**Comment by gmarull** [comment]
> > see my comment
> 
> I see it is merged now

Then please, align

**Comment by kl-cruz** [comment]
> > > see my comment
> > 
> > 
> > I see it is merged now
> 
> Then please, align

done

File: soc/nordic/nrf54l/Kconfig [hunk 23]
```diff
@@ -56,8 +67,15 @@ config SOC_NRF_FORCE_CONSTLAT
...
 	bool "NRF54L DC/DC converter."
+	select DEPRECATED
 	help
 	  To enable, an inductor must be connected to the DC/DC converter pin.
+	  This option is deprecated, use devicetree instead. Example
+	  configuration:
+
+	  &vregmain {
+	    regulator-initial-mode = <NRF5X_REG_MODE_DCDC>;
+	  };
```
Comment (gmarull) [inline soc/nordic/nrf54l/Kconfig:78]: I don't think there's a need to deprecate on nRF54L (still in engineering phase)

Comment (kl-cruz) [reply soc/nordic/nrf54l/Kconfig]: I updated code with dts-like approach

File: soc/nordic/nrf54l/soc.c [hunk 24]
```diff
@@ -149,7 +151,12 @@ static int nordicsemi_nrf54l_init(void)
 		nrf_power_task_trigger(NRF_POWER, NRF_POWER_TASK_CONSTLAT);
 	}
 
+#if defined(CONFIG_SOC_NRF54L_VREG_MAIN_DCDC)
 	if (IS_ENABLED(CONFIG_SOC_NRF54L_VREG_MAIN_DCDC)) {
```
Comment (hubertmis) [inline soc/nordic/nrf54l/soc.c:155]: What is the reason to keep this branch if the configuration options was removed?

**Review by hubertmis — DISMISSED** [review]

File: boards/nordic/nrf54l20pdk/board.cmake [hunk 25]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(jlink "--device=cortex-m33" "--speed=4000")
+
+include(${ZEPHYR_BASE}/boards/common/nrfjprog.board.cmake)
```
Comment (gmarull) [inline boards/nordic/nrf54l20pdk/board.cmake:6]: nrfjprog is not supported for this soc: cc: @carlescufi

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/nordic/nrf54l20pdk/board.cmake (hunk 25 above)
Comment (carlescufi) [reply boards/nordic/nrf54l20pdk/board.cmake:6]: Yep, please use only nrfutil @kl-cruz

File: soc/nordic/nrf54l/soc.c (hunk 24 above)
Comment (kl-cruz) [reply soc/nordic/nrf54l/soc.c:155]: I reverted this change and remove kconfig option in another PR

File: boards/nordic/nrf54l20pdk/board.cmake (hunk 25 above)
Comment (kl-cruz) [reply boards/nordic/nrf54l20pdk/board.cmake]: done

File: soc/nordic/nrf54l/Kconfig (hunk 23 above)
Comment (kl-cruz) [reply soc/nordic/nrf54l/Kconfig]: I will update L15 in another PR, to do not mix two tasks in one PR

File: soc/nordic/nrf54l/soc.c (hunk 24 above)
Comment (kl-cruz) [reply soc/nordic/nrf54l/soc.c:155]: see: https://github.com/zephyrproject-rtos/zephyr/pull/77160

File: soc/nordic/nrf54l/Kconfig (hunk 23 above)
Comment (kl-cruz) [reply soc/nordic/nrf54l/Kconfig]: see: https://github.com/zephyrproject-rtos/zephyr/pull/77160

File: soc/nordic/nrf54l/Kconfig.defconfig.nrf54l20_enga_cpuapp [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_NRF54L20_ENGA_CPUAPP
+
+config NUM_IRQS
+	default 274
+
+config IEEE802154_NRF5
+	default IEEE802154
```
Comment (gmarull) [inline soc/nordic/nrf54l/Kconfig.defconfig.nrf54l20_enga_cpuapp:12]: this should not be needed

File: soc/nordic/nrf54l/soc.c [hunk 27]
```diff
@@ -149,7 +151,15 @@ static int nordicsemi_nrf54l_init(void)
...
+#elif (DT_PROP(DT_INST(0, nordic_nrf5x_regulator), regulator_initial_mode) == \
+		NRF5X_REG_MODE_DCDC)
+	{
+#else
+	/* DCDC is not configured neither in KConfig nor DTS. Disable its configuration. */
+	if (0) {
+#endif
```
Comment (gmarull) [inline soc/nordic/nrf54l/soc.c:162]: please, split these changes into its own commit, also, `if (0)` is redundant, just ifdef out the whole block

File: dts/common/nordic/nrf54l20.dtsi [hunk 28]
```diff
@@ -0,0 +1,563 @@
...
+				interrupts = <270 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+
+			reg: regulator@120600 {
+				compatible = "nordic,nrf5x-regulator";
+				reg = <0x120600 0x1>;
+				regulator-name = "REG";
+				regulator-initial-mode = <NRF5X_REG_MODE_LDO>;
+			};
```
Comment (gmarull) [inline dts/common/nordic/nrf54l20.dtsi:525]: this doesn't describe regulators in nRF54L20 correctly, please, define vregmain as a child node (see some other nrf socs doing this)

File: dts/arm/nordic/nrf54l20_cpuapp.dtsi [hunk 29]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (gmarull) [inline dts/arm/nordic/nrf54l20_cpuapp.dtsi:1]: dt should come before soc content, otherwise soc Kconfig options can't be used at all (ref: keep things bisectable)

File: modules/hal_nordic/nrfx/CMakeLists.txt [hunk 30]
```diff
@@ -165,6 +167,10 @@ zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54LX_SKIP_CLOCK_CONFIG NRF_SKIP_C
 zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54LX_DISABLE_FICR_TRIMCNF NRF_DISABLE_FICR_TRIMCNF)
 zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54LX_SKIP_GLITCHDETECTOR_DISABLE NRF_SKIP_GLITCHDETECTOR_DISABLE)
 
+# Inject code to skip TAMPC setup for nRF54L20. It is not supported for now.
+# It needs to be removed when support is provided.
+zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF54L20_ENGA_CPUAPP NRF_SKIP_TAMPC_SETUP)
```
Comment (gmarull) [inline modules/hal_nordic/nrfx/CMakeLists.txt:172]: if nrf54l20 specific, it should be conditioned to it, doesn't seem to be the case

**Review by gmarull — CHANGES_REQUESTED** [review]

File: dts/common/nordic/nrf54l20.dtsi (hunk 28 above)
Comment (kl-cruz) [reply dts/common/nordic/nrf54l20.dtsi]: Updated to solution introduced in nRF54L15

File: dts/arm/nordic/nrf54l20_cpuapp.dtsi (hunk 29 above)
Comment (kl-cruz) [reply dts/arm/nordic/nrf54l20_cpuapp.dtsi]: done

**Review by gmarull — DISMISSED** [review]

**Review by jaz1-nordic — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]

File: dts/common/nordic/nrf54l20.dtsi [hunk 31]
```diff
@@ -0,0 +1,572 @@
...
+				compatible = "nordic,nrf-timer";
+				status = "disabled";
+				reg = <0xce000 0x1000>;
+				cc-num = <6>;
+				max-bit-width = <32>;
+				interrupts = <206 NRF_DEFAULT_IRQ_PRIORITY>;
+				prescaler = <0>;
+			};
+
+			pwm20: pwm@d2000 {
```
[the remark this answers is not in the document]
Comment (kl-cruz) [reply dts/common/nordic/nrf54l20.dtsi:353]: You might be right

File: boards/nordic/nrf54l20pdk/doc/index.rst (hunk 1 above)
Comment (kl-cruz) [reply boards/nordic/nrf54l20pdk/doc/index.rst]: Should we fix it in nRF54L15 DK's doc?

Comment (anangl) [reply boards/nordic/nrf54l20pdk/doc/index.rst]: I think so.

**Review by dchat-nordic — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
